target_snowflake/sinks.py

```diff
@@ -1,44 +1,140 @@
 """Snowflake target sink class, which handles writing streams."""
 
-from typing import Dict, List, Optional
+import datetime
+import tempfile
+import pytz
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional
+from typing import Any, Dict, List
 
-from singer_sdk.target_base import Target
-from target_snowflake.database_target.csv_sink import CSVSink
-from target_snowflake.migrator import SnowflakeSchemaMigrator
+from singer_sdk import SQLSink, SQLTarget
 
+from target_snowflake.connector import SnowflakeConnector
+from target_snowflake.stages import NamedStage
 
-class SnowflakeSink(CSVSink):
+import csv
+
+
+class SnowflakeSink(SQLSink):
     """Snowflake target sink class."""
 
+    connector_class = SnowflakeConnector
+    stage_class = NamedStage
+
     def __init__(
         self,
-        target: Target,
+        target: SQLTarget,
         stream_name: str,
         schema: Dict,
         key_properties: Optional[List[str]],
     ) -> None:
         super().__init__(target, stream_name, schema, key_properties)
+        self.prepare_sink()
+
+    @property
+    def table_name(self) -> str:
+        return self.stream_name.split("-")[-1].upper()
+
+    @property
+    def schema_name(self) -> Optional[str]:
+        """Return the target schema name."""
+        return self.config["snowflake"]["schema"].upper()
```

Comment:
According to `snowflake-sqlalchemy` we should probably actually use lower-case for these:

https://github.com/snowflakedb/snowflake-sqlalchemy/tree/9118cf8f18a0039f9cb5d3892ff2b1e5c82a05e0#object-name-case-handling

Reply:
@rabidaudio - Good catch! 👍